### ISSUE TITLE
Analyze status-go binary sizes

### ISSUE THREAD

**Issue by adambabik** [status-im/status-go#1133]
# Problem

There is a couple of use cases that require as smallest binaries as possible, for example, [Android Go](https://www.android.com/versions/oreo-8-0/go-edition/).

We should analyze status-go binary sizes:
- how do our binary sizes compare with geth binaries?
- is there anything we can do to make the binaries smaller?
- what's CGO overhead on the binary size, if any?

**Comment by divan** [comment]
So Go typically doesn't care much about large binaries size, because original target platform for Go were servers, and disk space is cheap there. However, there are constant attempts to lower the binary sizes or, at least, prevent growing it. The big binary size mostly comes from the fact that Go compiles in everything statically, including standard libraries and runtime code, and from debug information.

Debug information can be stripped of with `-ldflags "-w -s"` (see docs: 
https://golang.org/cmd/link) flags. It will affect debugging, profiling and stacktraces in case of panic (though they should remain readable).

I tried quickly assess the gains from stripping off debug information from both `statusd` and Android `statusgo` library:

 - `statusd` - 45170248 → 24564768 bytes
 - `statusgo lib` - 38287440 → 24428284 bytes

Also, I checked [upx](https://github.com/upx/upx) minifier, which essentially compresses the code in a smart way and decompress on init. This is not always working well for all platforms and cases, but worth checking anyway. I was able to reduce `statusd` binary for amd64 to 9MB, and `statusgo` library to 16.7MB. (for the Android build I had to use `devel` branch of `upx` tool to work properly).

So, bottom line:
```
statusd:
===============
Default build:      45170248 bytes
Without debug info: 24564768 bytes
UPX:                 9052176 bytes

statusgo lib for Android (upx devel):
===============
Default build:      38287440 bytes
Without debug info: 24428284 bytes
UPX:                16734534 bytes
```

So I see a lot of space for optimization for the binary size. We need to set more concrete objectives and experiment with different ways of reducing binary size further.

<blockquote><div><strong><a href="https://golang.org/cmd/link">link - The Go Programming Language</a></strong></div></blockquote>
